components/prism-factor.js

```diff
@@ -0,0 +1,344 @@
+(function (Prism) {
+
+	var comment_inside = {
+		bold: /\b(?:TODO|FIX|FIXME|NOTE|BUG|XX+|HACK|\?\?+|!!+)(?:\b|:)/
+	};
+	var string_inside = {
+		number: /(?:\\\S|%\w)/
+	};
+
+	var factor = {
+
+		'comment': [
+			{
+				// ! single-line exclamation point comments with whitespace after/around the !
+				pattern: /(^|\s)(?:! .*|!$)/,
+				lookbehind: true,
+				inside: comment_inside
+			},
+
+			/* from basis/multiline: */
+			{
+				// /* comment */, /* comment*/
+				pattern: /(^|\s)\/\*\s.*?\*\/(?=\s|$)/,
+				lookbehind: true,
+				inside: comment_inside
+			},
+			{
+				// ![[ comment ]] , ![===[ comment]===]
+				pattern: /(^|\s)!\[(?:\[\s.*?]|=\[\s.*?]=|==\[\s.*?]==|===\[\s.*?]===|====\[\s.*?]====|=====\[\s.*?]=====|======\[\s.*?]======)](?=\s|$)/,
```

Comment:
How about this:
(Assuming there can be any number of `=`)

```suggestion
				pattern: /(^|\s)!\[(=*)\[\s.*?\]\2\](?=\s|$)/,
```

Reply:
I didn't know browser javascript regex had backreferences either, but thanks, i've fixed it